### DISCUSSION TITLE
Training Effect Not Calculated

### DISCUSSION THREAD

**Discussion by ycazam** [philosowaffle/peloton-to-garmin#670] [TE / TSS / Vo2 / Intensity Minutes]
Hi all,

New User and I'm working with the following setup:
Garmin Forerunner 265
Peloton Bike+
Window App GUI

Looking for help on ultimately VO2 Max. But for now, Training Effect and Load were not calculated on my most recent ride on the Peloton, or the 10ish from before I selected to sync. Garmin Connect does show the training effect for any of the imported Peloton ride.

My method was to broadcast data from watch to bike, and also start a separate activity from the watch itself.

The resulting sync from P2G does provide great stats like HR, Power, Cadence, Speed, Intensity Minutes and even TSS/IF does calculate. But it is missing Training Effect and Load, and Temperature but that is less important. 

Am I missing anything? I also created an account on Tacx. I also tried to do the sync after changing the device info to the same that matches my native Garmin watch activity (except Manufacturer ID which I set to 1, couldnt find that). Convert to FIT and Upload to FIT are selected.

Was I not supposed to also start an activity from my watch?
Is there anyway to download the almost 100% perfect file from P2G and add/merge the missing data from my native Garmin activity?
Do I have to wait some time for it to be calculated? I waited a few hours and nothing changed.

Any help is appreciated.

**Comment by mcgonago** [comment]
I am in the same boat - I am also using the Peloton To Garmin (P2G) application to sync my activities to garmin.

Yes - everything is there except Training Effect.

Without Training Effect - Training Status and Load are not recorded.

I have read elsewhere that Garmin calculates Training Load using Firstbeat Analytics - which requires specific data (hardware?) from your watch.

This specific data is not pulled down by P2G for upload to Garmin connect.

I am pretty sure we are s%Ht out of luck here.

Any help is appreciated.

**Comment by mcgonago** [comment]
Thinking...

When on the Peloton - you can go to HR on your watch, select options, and hit Broadcast HR to Peloton.

Then - start a indoor bike activity at the same time you start Peloton workout...

Stop and save Garmin activity at end.

I "think" the Garmin will only have raw metrics for Training Effect - not all the other nice things the P2G includes in their .tcx files

Will try it...

**Comment by mcgonago** [comment]
Bug with P2G saved tcx files

You need to rename the TrackPoint (with capital P) to Trackpoint (lowercase P) before uploading to Garmin connect

**Reply by ycazam** [reply]
> Bug with P2G saved tcx files
> 
> You need to rename the TrackPoint (with capital P) to Trackpoint (lowercase P) before uploading to Garmin connect

Did that end up working for you? 

Do you now get Training Effect?

Are you also on Forerunner 265? I chatted Garmin support and they 'assured me' it wouldnt work with Tacx, and said it should work with Zwift. I havent yet tried to spoof the file to make it appear as such.

**Reply by mcgonago** [reply]
Tried using the watch "broadcast HR" - Peloton does see it...

As expected - Training Effect is not included in saved .tcx 

99% certain - the only way to get Garmin Training Effect score is to record activity from the watch

Garmin does not export (expose) its data for how it determines Training Effect (Firstbeat Analytics)

Though - I seem to recall Zwift rides that get uploaded to Garmin do have Training Effect. That fact seems
to indicate that Peloton could do whatever Zwift does when uploading activities to Garmin?

**Comment by mcgonago** [comment]
The change allowed all the graphs and tracking data to show up - except
Training Effect

I have read - the only way to get garmin training effect metrics is to
capture activity using (on) the watch

There is a chance - broadcasting HR to peloton - and then using p2g -
training effect may show up (with no graphs of other cadence, speed, power,
…)

Pushing activity to Strava through peloton app shows all data (Strava
generates its own training (relative) effect)

I would be ok with training effect only on Garmin (the training load
metrics on Garmin are my go to for fitness tracking)

Will let you know what I find

**Comment by philosowaffle** [comment]
Training Effect has been a difficult metric to solve in the past, it works for some people and not for others, I'm not sure we have ever isolated exactly why that is.  

For what its worth, TE and Load both show up without issue on my P2G synced workouts.

1. My personal watch is a Fenix
2. My HR zones are configured on Garmin connect
3. I dont use my watch at all on the Peloton, I just wear an HR monitor connected to the bike
4. P2G is configured to upload the FIT file
5. P2G is configured to use the default Tacx device
6. I do not have an account on Tacx

**Reply by mcgonago** [reply]
Thank you for the info

Good news - I am now seeing Training Effect in my Garmin activities

What changed?

I believe it was number (4) upload the FIT file (settings -> conversion -> Convert to FIT)

Also went to Settings -> Garmin -> Advanced -> Format to Upload -> FIT

After upload to Garmin - I did notice my first Tacx Ride badge for this mornings ride.

I also use the HR broadcasted from my Garmin watch (during ride, go to HR menu - hit select, and enable Broadcast HR)

I do not believe the HR needs to be broadcasted - I have a Garmin HR strap which I will probably move to (may be a bit more accurate)

Again - thx

**Reply by mcgonago** [reply]
One more thing...

When you say (5) P2G is configured to use the default Tacx device

Are you referring to the Cycling device in Settings -> Conversion -> Advanced -> Exercise type -> Cycling -> TacxTrainingAppWin

I did not change this (that was the default)

Is there somewhere else where Tacx is referenced?

Not a huge deal since Training Effect works now - just curious

**Reply by mcgonago** [reply]
A bit more info (again, I am all set - but, just in case you are not..)

I mentioned that my Garmin Activity shows up as a Tacx activity/device

I believe that comes from the setting of Settings -> Conversion -> Advanced -> Exercise type -> Cycling -> TacxTrainingAppWin

For kicks, I went into advanced, deleted all the entries, hit save

By default I then see the Forerunner 945 showing up (only)

I then do a re-upload of the activity - Garmin now thinks my activity was done using a Forerunner 945

Short story - the Advanced -> Exercise Type settings do not appear to affect Garmin activity metrics (Training Effect always shows up)

**Reply by ycazam** [reply]
Thank you thank you. Super helpful. 

I also got the Tacx Badge, will try tmrw during my ride to see what else I might’ve missed here and report bad.

**Reply by philosowaffle** [reply]
> Are you referring to the Cycling device in Settings -> Conversion -> Advanced -> Exercise type -> Cycling -> TacxTrainingAppWin

Yes, I'm referring to that.  I pretty much use all of P2G's default settings so I can validate that P2G works "out of the box" for most folks.

As you discovered, Garmin doesn't treat TCX and FIT files the same way which is why FIT is the default and recommended file format when uploading to Garmin.  

I've kept TCX around as a legacy option but in all cases you'll have more data and a better experience if you send FIT to Garmin.

**Reply by mcgonago** [reply]
Thanks for the info

I was playing around with FitSDKRelease_21.141.00.zip which allowed me to convert my .fit to .csv

I do see in the .csv much more info - with respect to "data definitions" for device time, ftp, power calculation type, ...

Will be interesting to see what @ycazam finds out...

**Reply by ycazam** [reply]
Thanks for sharing all your finding so far @mcgonago 

However I am a little confused on how you got it working exactly.
-Are you using TCX or FIT file?
-Are you manually uploading a file to garmin, or using the P2G sync feature?
-Is strava involved at all?
-Does Training Effect immediately appear, or is there a delay from when it hits Garmin, to allow for some calculations on the watch?
-does the peloton activity appear on your watch? 

What I am currently doing:
-I was already doing the Broadcast HR to Peloton trick, and it was lacking Training Effect (all other data/graphs still show, like HR, Power, Cadence, etc.)
-I am using Tacx as 'device' and syncing from P2G (I also tried deleting all entries as you mentioned to no avail)
-As a workaround, I am creating two activities for my one peloton workout: one activity is recorded on my watch only which does provide Training Effect, and a second one recorded on Peloton which has everything but Training Effect.

Much appreciated on your help so far.

**Reply by mcgonago** [reply]
Good questions 

The only way I got it to work was by selecting the two FIT options in P2G

I just do a manual sync from P2G to Garmin and all is good

I do select the option that also downloads the .fit file to my laptop - but, that is not necessary (since I hit the sync button)

Though, I did verify I could upload the .fit to Garmin - that also worked

Be careful to make sure FIT is enabled in the two places (previous reply details the places)

You should only need to broadcast HR from watch (no need to track workout)

I would try:
- remove saved activity on garmin
- re-sync your P2G activity
- see the .fit generated file

If this new sync still shows no training effect - delete it through the garmin connect app, and try manual upload of your .fit from garmin connect app (web interface)

Let me know how it goes

If all else fails, you could email me your .fit file - I can convert it to .cvs format and compare content against my working .fit file

Btw - Strava not involved at all, Training Effect score immediately appears

mcgonagleo@gmail.com

**Reply by ycazam** [reply]
Thanks for being so helpful. I do have FIT enabled in these two places:
![image](https://github.com/user-attachments/assets/c47308a6-ab42-4b70-8864-e95598e7dc34)
![image](https://github.com/user-attachments/assets/69db5615-0692-491e-8c21-f4dcc05048f0)

To confirm, you have the Forerunner 265, right?

Just tried as you suggested, including the 'save a backup copy' locally, and manually uploading that file -- still nothing under TE.

I just emailed you that FIT file. You can also send me your working version FIT File so I can also try to troubleshoot.

Much appreciated again.

**Reply by mcgonago** [reply]
I will take a look at your .fit

I use a forerunner 255s - not sure how much that would make a difference?

Here (not to rub it in - more about what you will eventually get) is a screen shot of all the good stuff


Will look at your .fit to see if I notice anything strange

![peloton_to_garmin_stats](https://github.com/user-attachments/assets/ad14229b-084f-4af3-a7f6-693af5ba3409)

**Reply by mcgonago** [reply]
I went for it - I uploaded your .fit and got these statistics

![martin_activity](https://github.com/user-attachments/assets/0b6d5469-f154-42c8-8b2a-f5ed4ece92b7)

**Reply by mcgonago** [reply]
which means your .fit file is fine

I did a side-by-side diff - and everything looks the same (outside of your ride having more data, and the raw numbers being different)

Now the question - maybe it has to do with a setting on my garmin connect application?

Thinking....

**Reply by mcgonago** [reply]
I guess you could try the same - upload my .fit that I emailed you - see if Training Effect shows up...

My guess - it will not show up - for (whaterver) the same reason(s) I get Training Effect and you do not.

**Reply by mcgonago** [reply]
grasping for straws

![upload_display_preferences](https://github.com/user-attachments/assets/c3bff944-e2a1-4c19-b16f-e3c55f2871e9)

**Reply by mcgonago** [reply]
Data settings

![data_settings](https://github.com/user-attachments/assets/93e9c777-c056-4a8a-8e32-5b639ac15c8d)

**Reply by mcgonago** [reply]
I would keep searching around...

The "tricky part" is how will you know, if you change a setting, that it will take effect?

My guess - any tweek you do, you will need to delete and re-load the .fit to see if the change takes affect?

One last clip - here is my FTP setting from my garmin connect app (web interface)

99% sure there is a setting that is not set - which is keeping the proper conversions from your .fit to show Training Effect...

![ftp](https://github.com/user-attachments/assets/f5d93228-427a-4f2e-bcf5-0e08990b06f3)

**Reply by ycazam** [reply]
Again, you have been a great help.

But, uploading your file did not work :/

Also updated my settings to match yours (I was on metric units and enabled Product Improvement) as well as privacy settings open to 'everyone'  -- still no luck after deleting and re-uploading:

![image](https://github.com/user-attachments/assets/ee4a6c13-5a47-46c9-b859-6efe29c51624)

I will keep investigating, and reach out to Garmin support to ask what the heck is up.

BTW... nice FTP, keep up the good work!

**Reply by mcgonago** [reply]
Thx on my FTP score

As you know - training effect and intensity minutes are missing for you...

Hmmm...

Wondering - do you have power meter pedals on your real bike?

I do see km in your screen shot not mi (doubt that is it)

We know this
- your .fit file is fine (works for me)
- my .fit does not work for you

All points to something with
1. A garmin connect setting

Which one? Not sure

I am having too much fun with this

I may try getting mine to break - find the magic setting

At this point - you should be able to share both mine and your .fit to garmin - "works for omcgonag and not me - what config setting is causing this?"

Not giving up quite yet

**Reply by mcgonago** [reply]
I did some googling - are your maximum heart rate and heart rate zones configured correctly?

![image](https://github.com/user-attachments/assets/87b3adbd-a067-49a4-bbf6-509763cbe38c)

**Reply by mcgonago** [reply]
This link has some good info

https://support.garmin.com/en-US/?faq=apva7Th82K6Mm0zY4shzD6

Slight chance you may need 2+ weeks of consistent rides

Along with some vo2 max results

Patience? 

Owen

**Reply by mcgonago** [reply]
Along with some outdoor rides from your watch that help kick things into gear

**Reply by mcgonago** [reply]
Maybe go for a short 0.2 mile jog - I know Garmin puts emphasis in vo2 max comparisons between running and cycling

**Reply by mcgonago** [reply]
Do your rides trigger (show up) in the v02 max training display on your phone garmin app?

If not - that would be the problem

**Reply by mcgonago** [reply]
Not to share too much info - if you do not see v02 max numbers - that would need to be fixed somehow

![image](https://github.com/user-attachments/assets/8cf49829-26a3-45f6-a2d1-e1f93163556e)

**Reply by mcgonago** [reply]
Lastly - you need heart rate and power zones setup on your garmin device - through Garmin connect app - select device and user settings

**Reply by mcgonago** [reply]
Last little quip...

Playing around - I noticed the "training effect" and "intensity minutes" seem to go together 

Also - after uploading .fit - the training effect shows up instantly, intensity minutes takes a bit

Intensity is probably a secondary calculation using power and heart rate zones

If your heart rate and/or power zones are off, or not set, that would explain everything

**Reply by mcgonago** [reply]
@ycazam - how goes it? Did u figure out how to get Intensity metrics?

**Reply by ycazam** [reply]
Hey, I have been luck with some warm temps outside, and have learn to dress for the cold so I havent needed to get back on the trainer until later today... Thanks for following up bud!

To respond of some of the thoughtful points you made:

- No, I do not have power meter pedals on your real bike. Only the Peloton which is built in.

- I think my maximum heart rate and heart rate zones configured correctly? I had two HRs set (one default and a lower one for Bike Activities). I have since set them to match my lower bike Max HR (181)

- I have no vo2 max results, likely from not having a Power Meter on my actual bike, and not doing any running which can also measure vo2 (although not sure if both exercises have their own vo2 caluclations) Based on your screenshots, it seems you have one from Running and one from Cycling?

- HR Zones are based off of my Percent of Max HR and using the default I believe.. 50%-60%/60-70/70-80/80-90/90%-100%

- Does LTHR matter at all? I set it to what I estimate to be mine, at 144.
 
- Power Zones are set up correctly I think, based off an FTP test I do around monthly on the PTon, (last test was 223W, and they are on the default zone settings). I havent touched Running or XC Ski Power Zones also in that menu area.

I gather you have Power Meters on your outdoor bike? That might be the crux of the issue, as it sounds everything else you have setup I am reasonably configured too.

I have not reached out to Garmin, but as I will be ramping up indoor training in the next few weeks, that is my next step.

**Reply by ycazam** [reply]
And based on the article you [shared](https://support.garmin.com/en-US/?faq=apva7Th82K6Mm0zY4shzD6&identifier=886785&tab=topics).. I think you are onto something:

> 
"Before your Garmin Connect™ account can analyze Tacx or third party activities, a cycling activity needs to be uploaded from a Training Status capable device **and a VO2 Max must be established**. After an activity has been uploaded from a training app, you will need to sync your training status device with your Garmin Connect account for Training Status to update on the Garmin device then sync back to your account."

Well shucks. How can I do that without a power meter on a bike? Borrow someone? Or maybe I can spoof it some other way.

Sorry if this was previously mentioned on the thread or in the documentation, feel like I wasted all your time here now.

**Reply by ycazam** [reply]
Also, I do have training status working fine. 

Wonder if by training my running and establishing a vo2 in that exercise, if it will carry over and allow for the cycling one to kick in and work like it does for you.

Guess I am running soon to give it a shot...

**Reply by mcgonago** [reply]
I think in summary a few things are important
- garmin needs 2+ weeks of activities that include v02 measurements 
- I got the v02 from runs and zwift rides

I do not have power pedals

Zwift power is pretty accurate

My guess? A few runs will help

Not sure if you have tried zwift - ton of fun

**Reply by ycazam** [reply]
How do you use Zwift? From what I see it does look fun.

Can I use my PTon for it? 

What is your setup like?

**Comment by DinoChiesa** [comment]
It works for me, like @philosowaffle describes.  

1. My watch is a Forerunner 965, but I think that is irrelevant because it is not involved in the data exchange.
2. My HR zones are configured (auto computed based on HRmax) on Garmin connect
3. I don't use my watch at all on the Peloton, I just wear an HR monitor connected to the bike
4. P2G is configured to upload the FIT file
5. P2G is configured to use the default Tacx device (or, no change to its default. In fact I could not find how to change the device, in the config file - I am using p2g_console, not the UI version)
6. I do not have an account on Tacx

I am using p2g_console v4.3.0 

And after I upload the fit file successfully, I can see the VO2max stats. And training effect.

**Reply by ycazam** [reply]
By chance, do you have a power meter on your outdoor bike, and/or have you gotten a VO2 Max previous while on a cycling or running activity outside?

**Reply by anlesk** [reply]
got the exact same setup and can also see training effect in Garmin.

@ycazam I have pedal-based PM on my outdoor bike (Assioma Favero) and had my VO2Max changed in the past based on those.
Also using footpod (Stryd) for running which calculates running VO2MAX

**Reply by DinoChiesa** [reply]
> do you have a power meter on your outdoor bike, and/or have you gotten a VO2 Max previous while on a cycling or running activity outside?

I do not.

**Comment by RJHug00** [comment]
I use a different [VBA] implementation to build my FIT file, but I get a computed Training Effect when the FIT *appears* to have originated from a legitimate Garmin watch.  I never found a way to get Training Effect computed without the FIT containing a legitimate Garmin device (the FIT uploaded, just no TE calculation).  I identify the device as a retired VENU2, complete with its actual serial number, that I have associated with my Garmin account (but it is no longer my primary wearable).  I make Peloton record heart rate from my Garmin chest strap during the ride so it comes back in the Peloton download along with power and cadence.
![Untitled](https://github.com/user-attachments/assets/51fe2ee0-956b-42d4-9a4e-f6ccb87b7fb9)
I never got around to experimenting to see if I could make the FIT appear to come from my [primary wearable], which I'm guessing is the boat most people are likely in.

**Reply by RJHug00** [reply]
Oh, and yes, I wear my primary Garmin watch while on the Peloton.  The watch often warns me I'm experiencing a period of high stress during the ride because I don't tell it I'm doing any particular activity.  I've never experimented to see what Garmin would do if my primary watch recorded a ride while I was also claiming that another watch recorded the same activity [with power and cadence].  You can see from this image that its already a bit confused about the stress it detected and when the exercise made stress measurement worthless.
![Untitled2](https://github.com/user-attachments/assets/4844d85a-774a-494d-8f2e-7f846b46ed3d)
For my purposes, what I have it doing is good enough.

**Reply by ycazam** [reply]
By chance, do you have a power meter on your outdoor bike, and/or have you gotten a VO2 Max previous while on a cycling or running activity outside?

**Reply by mcgonago** [reply]
@ycazam - are you still having problems seeing TE?

**Reply by RJHug00** [reply]
@ycazam - I only ride the Peloton Bike+ [indoors] with its built-in power sensor, and my only VO2 Max tests were done indoor on a treadmill. Not sure what that tells you; not entirely sure your questions were actually posed to me.

**Reply by DinoChiesa** [reply]
> I only ride the Peloton Bike+ [indoors] with its built-in power sensor

same

**Comment by jeetshah26** [comment]
@ycazam did you ever get this to work. I too do not bike outdoor and started using a peloton to get some variety in workouts ,but i cannot get any Training status or load to be calculated which means the watch always disregards peloton in load :(

**Reply by RJHug00** [reply]
@ycazam - The pic below shows my Peloton rides of the last week included in Training Load.  As discussed before, the rides' FIT files **_must_** appear to originate from a Garmin device for this to happen. I have never experimented to see if it works if the FIT file identifies my normal primary wearable; I always identify a different watch that it also associated with my Garmin login.

![TrainingLoad](https://github.com/user-attachments/assets/2a227a4e-2030-4ead-9d0a-8b60046c57f5)

**Reply by jeetshah26** [reply]
@RJHug00 interesting. I am changing my device to use my current epix device id. hopefully i start getting the data soon. i used the device id by exporting an existing workout and peeking at the tcx

**Reply by jeetshah26** [reply]
Well, that causes an interesting problem. if i use my current watch, it still doesnt calculate anything, but my watch also does not then sync this activity and the counts on my primary watch for weekly stuff is off. i'm going to stop chasing this and just let it use tacx.
I even went to the extent of deleting the activity from connect and putting the fit file on my device and no results. and unfortunately i dont have another garmin device on my account to setup with this.

**Reply by RJHug00** [reply]
Can't say that I blame you for giving up.  It shouldn't be this difficult, and yet it is.

My Garmin award for stupidity still goes to the stair stepper activity, which provides no means anywhere for entering the number of floors climbed.

**Comment by yetdog** [comment]
So - interesting occurrence here. Love this past discussion - helping me to understand a ton - so thank you!

Situation:
- Installed P2G, all good there.
- Bought and setup Epix Gen 2 TODAY
- Updated P2G to use my real device ID

I went through the setup for my peloton ride today, whereas I recorded an indoor bike activity on the watch, and then discarded it after I was done. I then used P2G to upload the ride, and sure enough it shows the training effect as hoped/expected.

However, I then went to sync a Peloton ride from 2 days ago, when I still did not have the Epix watch. It uploaded fine, but no Training Effect info attached!!

I exported and compared both .FIT files, and they're practically identical in structure/data except for the actual trackpoints. But the trackpoints themselves have the same metrics in both files. The file from today's ride (while wearing watch) didn't seem to contain any unique markers vs the one from two days ago, before I had the watch.

Could Connect be talking BACK to the watch and confirming that the watch was "alive/active" during an activity? From what I gather, TE is actually calculated on-device, so only rides that happened while I was actually wearing the device will calculate those metrics? Does that make sense?

Thanks y'all.

**Reply by RJHug00** [reply]
Interesting.  I never actually sync the VENU2 identified in my Peloton FIT files - haven't even turned it on for over a year, so GC isn't asking that watch if it was alive at the time.  I can say that TE shows up on GC Web almost immediately after uploading the FIT, which is almost always before I synch my primary wearable EPIX2 (which I was wearing, but was NOT recording an activity).  I had come to the conclusion that TE is computed on the servers now, at least for our type of Peloton FIT activities.  I even did some experiments of putting imaginary aerobic and anaerobic TE into the FIT file, and seem to recall Garmin ignoring those values.  Those experiments were over a year ago, so things might have changed in that respect.

**Reply by yetdog** [reply]
> Interesting. I never actually sync the VENU2 identified in my Peloton FIT files - haven't even turned it on for over a year, so GC isn't asking that watch if it was alive at the time. I can say that TE shows up on GC Web almost immediately after uploading the FIT, which is almost always before I synch my primary wearable EPIX2 (which I was wearing, but was NOT recording an activity). I had come to the conclusion that TE is computed on the servers now, at least for our type of Peloton FIT activities. I even did some experiments of putting imaginary aerobic and anaerobic TE into the FIT file, and seem to recall Garmin ignoring those values. Those experiments were over a year ago, so things might have changed in that respect.

Hmmm. But you DID have the VENU2 attached to your connect account ahead of time. I suppose I could test it going forward by doing a peloton ride tomorrow without the watch, and see if that does the TE stuff. If so, it would seem "has an appropriate garmin device linked in connect" might be the go/no go piece.

Another wrinkle, tho, was I have an edge 1040 thats also linked that i used once back in May, but having that there doesn't allow my past peloton rides to calc TE. This is all so exhausting.

The identical structure in a non-TE FIT file vs a TE FIT file is really the flummoxing part. theres nothing between the two that would indicate "yeah, this should calculate TE". So it certainly is something on the connect side.

**Reply by RJHug00** [reply]
Yes, the VENU2 is what got me into this ecosystem, and what had been recording run activities back in the day.  Its in the junk drawer now, but It remains identified on my Garmin account with the newer EPIX2 as my primary wearable.

Lots and lots of this is flummonxing; I don't understand what Garmin gains by keeping so many of the details secrets we have to uncover by all manner of exhausting experimentation.  I don't know enough about the Edge 1040 to know if it could be a viable source of a FIT ride.

Something else I noticed in my notes from experiments of yesteryear - the TE calculation seems to require the FIT file (from Peloton) to contain heart rate data.  I certainly don't remember the details of the experiment that led to that conclusion, but I do know that I have had Peloton record HR from my chest strap ever since.

**Reply by philosowaffle** [reply]
This is starting to sound consistent with how Garmin (in my opinion) [effectively paywalls VO2 Max](https://github.com/philosowaffle/peloton-to-garmin/discussions/654#discussioncomment-10306331).  From other threads of research we arrived at a similar conclusion that Garmin does do most of these calculations server side (not on the device itself as they claim) and that they only do the calculation if they think you own a particular level of device.

Unless of course, Garmin is sending these uploaded files to the personal device to get a calculation that is then sent back to the server with the result.  But that sort of design choice would be surprising to me.

**Reply by RJHug00** [reply]
In my specific case, I can state categorically that GC is not sending the file back to a device for the calculation [but it does eventually send the activity to my primary wearable for inclusion in other displays]. The TE shows up server-side on GC web essentially immediately after upload and I do NOT maintain a continuously-connected watch through the Garmin app on my phone.  Indeed, I only connect the watch to GC via the app once or twice per day.

**Reply by RJHug00** [reply]
And, specifically relative to VO2Max, its hysterical to me that Garmin thinks it worth paywalling, because it is SOOOOO wrong.  In at least one forum post, I detail what the watch has been telling me and then what two separate professional lab treadmill tests computed.  Off by roughly 20 percent.  Don't get me started on how wrong the watch can be for heart rate.  LMAO

**Reply by RJHug00** [reply]
But contrary to my last, we all know if you record a run or other activity on the watch, you can see the TE in the summary statistics on the watch before ever uploading it, *and* I know there are records/fields in the FIT file for the device to send *its* computed values to GC.  The computation can be done both places, apparently.  YET, as I noted elsewhere, there's some special secret handshake for whether GC uses the TE values from the FIT or computes its own values.

**Comment by DinoChiesa** [comment]
Not sure if this is relevant, but ... there is an odd thing I see when using the p2g_console app to synchronize .... the time on the ride seems to be dated _in the future_.  So any ride I sync this way  doesn't actually show up until ... sometime "tomorrow".  I am not sure if that is related to TE.    The TE still works for me.

**Reply by mcgonago** [reply]
I do not see this when I sync an activity - shows up right away

Maybe a time sync issue with your computer?

How many out there still have issues with TE not showing up?

Owen

**Reply by mcgonago** [reply]
I use the UI version

**Reply by DinoChiesa** [reply]
The computer running the p2g_console app... runs in UTC time zone.  So that may be related.

**Reply by RJHug00** [reply]
I upload the FIT through the web UI, and it shows up on the web dashboard almost immediately, 'labeled' with the correct local time that I actually began the ride.  Computed TE is in the activity when I click on it through the web UI - not like you must wait overnight.  I don't see the activity in the GC app or on my watch until the next time I sync the app, of course.   @DinoChiesa - what timestamp do you see if you run a dump utility on the FIT file?  runalyze dot com shows me a TYPE=0 creation_time in UTC that is properly offset for my local time and DST.

**Reply by DinoChiesa** [reply]
Thanks for that. I'll have to look again. It hasn't even bothering me so much that I feel the need to track it down further.

**Comment by Grafvonsteiner** [comment]
I have a Garmin Edge 540 but do not own a watch and I dont get any Training Effect data or calculations.
I just want to know if I understood the prior discussions correctly: Is the consensus that you need to own a Garmin watch to get Training Effect data and so on?
And the same with VO2Max: To get my synced Peloton workouts to count for the VO2Max, I first need to ride outside with a power meter?
